libs/extensions-registry/tinymce-plugin.model.ts

```diff
@@ -0,0 +1,13 @@
+import type { ClassConstructor } from "@umbraco-cms/backoffice/models";
+import { TinyMcePluginArguments, TinyMcePluginBase } from "../../src/backoffice/shared/property-editors/uis/tiny-mce/plugins/tiny-mce-plugin";
+import type { ManifestBase } from "./models";
+
+export interface ManifestTinyMcePlugin extends ManifestBase {
+	type: 'tinyMcePlugin';
+	meta: MetaTinyMcePlugin;
+}
+
+export interface MetaTinyMcePlugin {
+	api: ClassConstructor<TinyMcePluginBase>;
```

Comment:
This can't really load through an "umbraco-package.json" file. We need to ignore this property for the JSON schema output (add a comment with an `@ignore` line above) and make sure that the `js` property to specify a file works.

Reply:
I've revisited this, and changed the plugin model to require `js` and `exportName`, and remove `api` entirely - instead, I'm now importing the reference module, then getting the class via the exportName, then instantiating the set of retrieved plugins. Means the manifest should now play nice with package json files, but does mean an additional magic-string-like property.

TinyMCE doesn't appear to play overly nicely with asynchronous plugin loading, so I have to import early, then instantiate each plugin slightly later as they require the editor instance as a constructor argument.

I can't see any way around it, but open to other suggestions.

Reply:
Take a look at the function `createExtensionClass` which allows loading a class extension through the "js" and/or the "loader" property, which you can also inherit from manifest base classes. It will by default give you the default export of the es module, which I think is perfectly reasonable to expect developers to include in their extensions.